blog/en-us/download.md

```diff
@@ -5,7 +5,17 @@
 
 > Release Notes: https://github.com/apache/incubator-dubbo/releases
 
-### 2.6.4 (latest version)
+### 2.7.0 (latest version)
```

Comment:
2.7.0 is not released yet.

And the link should be updated too.

Once a release is performed, the final links should be something like this:

https://dist.apache.org/repos/dist/release/incubator/dubbo/2.7.0/apache-dubbo-incubating-2.7.0-source-release.zip

The following link is just a temporary link just for the vote purpose.

https://dist.apache.org/repos/dist/dev/incubator/dubbo/2.7.0/apache-dubbo-incubating-2.7.0-source-release.zip